tools/src/replaceImportMetaUrl.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// eslint-disable-next-line node/no-extraneous-import
+import {smart} from '@babel/template';
+import {Statement} from '@babel/types';
+import {Visitor} from '@babel/core';
+
+export interface PluginOptions {
+  opts?: {
+    replacementValue?: string;
+  };
+}
+
+export default function replaceImportMetaUrl(): {
+  visitor: Visitor<PluginOptions>;
+} {
+  return {
+    visitor: {
+      CallExpression(path, state) {
+        const opts = state.opts || {};
+        const replacementValue = opts.replacementValue || '__dirname';
+        const {node} = path;
+        if (
+          node.callee.type === 'MemberExpression' &&
+          node.callee.object.type === 'Identifier' &&
+          node.callee.object.name === 'path' &&
```

Comment:
@sofisl can we remove the `path` portion of this check? I noticed that depending on compiler settings this import name might get mangled to something else.

Reply:
Ignore the above. However, for this to not get mangled I had to do `import * as path from 'path';` as opposed to `import path from 'path';`

Reply:
@ddelgrosso1, we could remove lines 37 and 38, and still preserve matching (tests still behave the same). Should we do that, that way it's not so fragile?